src/material/badge/badge.ts

```diff
@@ -204,6 +188,25 @@ export class MatBadge extends _MatBadgeBase implements OnInit, OnDestroy, CanDis
       this._updateRenderedContent(this.content);
     }
 
+    if (typeof ngDevMode === 'undefined' || ngDevMode) {
+      const nativeElement = this._elementRef.nativeElement;
+      const matIconTagName: string = 'mat-icon';
```

Comment:
The type here can be trivially inferred by TypeScript
`const matIconTagName = 'mat-icon';`